pgcli/pgexecute.py

```diff
@@ -11,6 +12,9 @@
 psycopg2.extensions.register_type(psycopg2.extensions.UNICODE)
 psycopg2.extensions.register_type(psycopg2.extensions.UNICODEARRAY)
 
+# When running a query, make pressing CTRL+C raise a KeyboardInterrupt
+psycopg2.extensions.set_wait_callback(psycopg2.extras.wait_select)
```

Comment:
Updated the PR with link to the article.

The function can be found [here](https://github.com/psycopg/psycopg2/blob/df9fbc515d6f586e84d7853965f2bc544177ac6a/lib/extras.py#L566). It is the same function as the one from the article, minus handling KeyboardInterrupt.